### DISCUSSION TITLE
Adding a PENDING_STATUS status for tickets

### DISCUSSION THREAD

**Discussion by samsplunks** [django-helpdesk/django-helpdesk#1155] [Ideas]
Hi All,

I would like to introduce a pending status for tickets.
This is used when the support team needs some feedback from the ticket creator and to remove idle time for support team work time calculations.

Usually, there are "pending" and "resumed" states in ticketing systems.
To have a minimal impact on the code, I am considering only adding a PENDING_STATUS and have the ticket move back to OPEN_STATUS once the ticket creator has replied (in Ticket.save()).

I would consider the PENDING_STATUS being part of OPEN_STATUSES for Ticket._can_be_resolved() and also for feeds.py and staff.py views.

For the escalate_tickets.py command I would not consider PENDING_STATUS tickets to be escalated.

What are your thoughts on this?
Any other mechanism I am not aware of?

**Comment by samsplunks** [comment]
To have the ticket move back to OPEN_STATUS once the ticket creator has replied, I think I would use signals outside of the helpdesk app as it is a very specific request.

**Comment by Benbb96** [comment]
There is already a field on the Ticket model : [on_hold](https://github.com/django-helpdesk/django-helpdesk/blob/147151a271da818c2a9fdd7c7eed960086bd4167/helpdesk/models.py#L528) which I think was created for this purpose.

But I agree with you, maybe it needs some refactoring in order to have this in the status.

**Reply by samsplunks** [reply]
From what I have read, the "on hold" boolean is used to forbid the escalation script to escalate on hold tickets.

My need is to remove the part where the ticket is bounced back at the creator for feedback from the time calculation of the support team effort.

I think I will propose just adding the PENDING_STATUS to the Ticket model and making sure it follows the OPEN_STATUS behaviour throughout the app.

**Comment by samsplunks** [comment]
And also make the STATUS_CHOICES customizable like PRIORITY_CHOICES.

`STATUS_CHOICES = helpdesk_settings.TICKET_STATUS_CHOICES`

**Comment by samsplunks** [comment]
For completeness, a PR was made to introduce this feature.

https://github.com/django-helpdesk/django-helpdesk/pull/1157

**Comment by uhurusurfa** [comment]
PR solves this discussion.
